common/src/main/java/com/wynntils/utils/render/type/AbstractTexture.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright © Wynntils 2024.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.utils.render.type;
+
+import net.minecraft.resources.ResourceLocation;
+
+public interface AbstractTexture {
```

Comment:
I've been thinking about some map/poi only texture interface/class/enum for a long time. However, I think this should be done backwards. Imo, there should be a `AbstractMapFeatureTexture` class, that is either a `WrappedTexture` (an icon that is defined in the `Texture` enum) or a `MapIconTexture` (a wrapper for `MapIcon`s). What do you think?

While we are here, we need to handle user loaded textures too. I don't think we impose any limits on them, which is bad. We should limit them to a 1:1 aspect ratio, with 16x16, 32x32 and maybe 64x64 pixel textures.

Reply:
Ok, I should have commented a bit what I was trying to do here. The AbstractTexture and icon wrapper was only intended as a temporary measure, which should be removed before integrating `mapdata-ports` into main.

Reply:
The better solution would have been to fix `Models.Marker.USER_WAYPOINTS_PROVIDER.addLocation()` so it did not require a Texture, but an `iconId` or something like that. But that likely requires aligning the current waypoint poi system better with MapData, which is a undertaking on it's own. (It is like a partial, but different, implementation of MapData)